### ISSUE TITLE
`Set-Content -Confirm` confirms twice

### ISSUE THREAD

**Issue by iRon7** [PowerShell/PowerShell#24434]
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://github.com/PowerShell/PowerShell/releases)
- [X] Search the existing issues.
- [X] Refer to the [FAQ](https://github.com/PowerShell/PowerShell/blob/master/docs/FAQ.md).
- [X] Refer to [Differences between Windows PowerShell 5.1 and PowerShell](https://learn.microsoft.com/powershell/scripting/whats-new/differences-from-windows-powershell).

### Steps to reproduce

Not sure whether this qualify a a bug, but if I use `Set-Content -Confirm` I need to confirm twice:
* on the operation **"Set Content"**, and on
* on the operation **"Clear Content"** which is in fact an implicit operation

Although, there are several workarounds available, this behavior is quite disturbing when replacing the content of multiple files.

```PowerShell
Set-Content -Value 'Foo' -Path .\Temp.txt
Set-Content -Value 'Bar' -Path .\Temp.txt -Confirm
```

### Expected behavior

> ```PowerShell
> Confirm
> Are you sure you want to perform this action?
> Performing the operation "Set Content" on target "Path: .\Temp.txt".
> [Y] Yes  [A] Yes to All  [N] No  [L] No to All  [S] Suspend  [?] Help (default is "Y"): y
> ```

### Actual behavior

> ```PowerShell
> Confirm
> Are you sure you want to perform this action?
> Performing the operation "Set Content" on target "Path: .\Temp.txt".
> [Y] Yes  [A] Yes to All  [N] No  [L] No to All  [S] Suspend  [?] Help (default is "Y"): y
> 
> Confirm
> Are you sure you want to perform this action?
> Performing the operation "Clear Content" on target "Item: .\Temp.txt".
> [Y] Yes  [A] Yes to All  [N] No  [L] No to All  [S] Suspend  [?] Help (default is "Y"): y
> ```

### Error details

```console
Note the second
```

### Environment data

```powershell
Name                           Value
----                           -----
PSVersion                      7.4.5
PSEdition                      Core
GitCommitId                    7.4.5
OS                             Microsoft Windows 10.0.19045
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

**Comment by mklement0** [comment]
Definitely a bug - it seems that (the implementing class behind) `Clear-Content` is called behind the scenes, which triggers the extra prompt.

Not only should the `Clear-Content` prompt not show, if you respond with `N`, the file is still modified, but the `-Value` is _appended_ to the existing file (that is, you then get `Add-Content` behavior).

**Comment by iRon7** [comment]
> it seems that (the implementing class behind) `Clear-Content` is called behind the scenes

Yes, this appears also from invoking `Set-Content` on a directory:

```PowerShell
Set-Content .\Test ''
Set-Content: Unable to clear content of '.\Test' because it is a directory. Clear-Content is only supported on files.
```

**Comment by rkeithhill** [comment]
WG-Engine reviewed this and agree this is a bug that should be investigated and fixed.

**Comment by iRon7** [comment]
It is still unclear to me why issues marked as <kbd>Up-for-Grabs</kbd> are triggered to be closed...

**Comment by doctordns** [comment]
My view is that this should NOT be the case. That said, something open for grabs that is not grabbed is not a great state to be in either.

Trigging after a week is wrong, I think we';d agree./  But after 6 months?: After a Year? If no one has taken up the opportunity, after a time, I'm not sure how much sense it makes to leave it open.

**Comment by mklement0** [comment]
>  I'm not sure how much sense it makes to leave it open.

Echoing @iRon7's sentiment in different words: The sensible approach is to leave `Up-for-Grabs` issues open _indefinitely_, so that community members - at whatever point in the future - can _discover them_, know that they've been green-lit for implementation, and therefore potentially submit PRs that may eventually provide a fix.

The usefulness of an approved fix / improvement has no expiration date.
